### ISSUE TITLE
Use container autowiring

### ISSUE THREAD

**Issue by leofeyer** [contao/core-bundle#799]
Use container autowiring once there is a beta version of Symfony 3.3.

**Comment by Toflar** [comment]
I am 👎  on this. Autowiring is a great addition for developing **applications**. Bundles should **never** rely on autowiring. If you meant the autoconfiguration feature, that's a different story.

**Comment by leofeyer** [comment]
Actually I mean both. :)

http://symfony.com/blog/new-in-symfony-3-3-service-autoconfiguration

http://symfony.com/doc/master/service_container/autowiring.html

Where did you read that bundles should not rely on autowiring?

**Comment by Toflar** [comment]
I did not read, it's obvious to me. Autowiring is always magic to some extent. I think they did a great job in making stuff as explicit as possible but it's still autowiring. Autowiring enables you to get to quicker results where you control **all** of the dependencies. That's only true when you are working on an **application**. A bundle, however, is imho not supposed to use any kind of autowire features.

**Comment by leofeyer** [comment]
But where is the point in writing this

```yaml
    contao.image.image_sizes:
        class: Contao\CoreBundle\Image\ImageSizes
        arguments:
            - "@database_connection"
            - "@event_dispatcher"
            - "@contao.framework"
```

when you can write this

```yml
    contao.image.image_sizes:
        class: Contao\CoreBundle\Image\ImageSizes
        autowire: true
```

?

**Comment by Toflar** [comment]
Autowiring is a RAD feature, it's not made for public bundles.

**Comment by aschempp** [comment]
Totally agree with @Toflar

**Comment by ausi** [comment]
> Where did you read that bundles should not rely on autowiring?

From the [Symfony documentation](http://symfony.com/doc/current/service_container/autowiring.html) (the last sentence):

> keep in mind to not use autowiring in public bundles nor in large projects with complex maintenance needs.

In your example, if I would register a second `@database_connection` service that implements the same interfaces, the autowiring would fail with `Unable to autowire...`.

**Comment by leofeyer** [comment]
Ok. Then just auto-configuration.

**Comment by leofeyer** [comment]
I have implemented auto configuration for commands in #804.

**Comment by leofeyer** [comment]
Implemented in 491502964b99b685af20d12564f7ba95102dd4d6.
